boofuzz/primitives/fuzz_list.py

```diff
@@ -0,0 +1,125 @@
+import random
+import glob
+
+from .base_primitive import BasePrimitive
+
+
+class FuzzList(BasePrimitive):
```

Comment:
It looks like the major added functionality in mutate is:
 1. If size parameter is enabled, skip oversized values.
 2. If size parameter is enabled, pad undersized values.

These could be moved to pre-filters, kind of like you did with `max_len`. If you do that, you can drop the custom mutate function and have a really simple class like `Delim` . What do you think?